### PR TITLE
Project exporter tool script

### PR DESCRIPTION
**Pull request by BereGabor**
Add helper script to exort git repost.
Walk in folder structure and export git repos from project folders.

### CODE REVIEW

File: export_repos.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        for gd in os.listdir(abs_dir):
+            abs_gdir = os.path.join(abs_dir, gd)
+            if os.path.isdir(abs_gdir):
+                #create target folder
+                target_dir = os.path.join(output_folder, d, gd)
+                os.makedirs(target_dir)
+                #call generator export to tmp folder
+                cli_line = format(PYTHON_COMMAND % (d, abs_gdir, tmp_output))
+                print(cli_line)
+                os.system(cli_line)
```
Comment (vifactor) [inline export_repos.py]: this does not look like a right thing to do: you are already in python interpreter, why would you launch others in such weird way?

Comment (BereGabor) [reply export_repos.py]: Should i restructure the code?

Comment (vifactor) [reply export_repos.py]: So you want the analysis to be done not on a single repo, but on a set of repos found in some location. Then add such option(s) to cli interface of the repostat's lauch script (for now it is gitstats) and simply call run-method on each repo found.

If you are not satisfied with performance (using os.system you perhaps can launch several interpreters in parallel), then consider using multiprocessing package.

One more possible way to go is to create your own separate project with the export_repos.py script, add repostat as a submodule and keep the technique you use now.

Comment (BereGabor) [reply export_repos.py]: Thanks. I resolved this issue, and call directly GitStat run. But i can import only if gitstat file have .py exctension. I pushed the commit. Could you check please?

**Comment by BereGabor** [comment]
Change request done, commited. Check it please.

File: export_repos.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    files = os.listdir(tmp_output)
+
+    for f in files:
+        shutil.move(os.path.join(tmp_output, f), target)
+
+print("Project folder: " + project_folder)
+print("Output folder: " + output_folder)
+
+#delete output dir
+try:
```
Comment (vifactor) [inline export_repos.py]: Please, wrap the script into 
```
if __name__ == '__main__':
    <your launching code>
```

File: export_repos.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                target_dir = os.path.join(output_folder, d, gd)
+                os.makedirs(target_dir)
+                #call generator export to tmp folder
+                g = gitstats.GitStats()
+                g.run(['-coutput=csv', format("-cproject_name=%s" % d), abs_gdir + ' ', tmp_output + ' '])
+                #move result csv from tmp folder to target dir
+                move_csv(target_dir)
+                #shutil.move(os.path.join(tmp_output, "*.csv"), os.path.join(target_dir, "*.csv"))
+
+try:
```
Comment (vifactor) [inline export_repos.py]: This is repeated twice, so better wrap into a method and at least print a message on exception

File: export_repos.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import sys
+import os
+import shutil
+import gitstats
```
Comment (vifactor) [inline export_repos.py]: Please, recheck if it gets imported, I wonder if we need to to
`import .gitstats`

Comment (BereGabor) [reply export_repos.py]: Hi I try in win and linux OS too, but doeasn't works this import without .py ext.

File: export_repos.py (hunk 2 above)
Comment (BereGabor) [reply export_repos.py]: Done. I restructure to OOP design

**Pull request merged**